vllm/model_executor/model_loader/weight_utils.py

```diff
@@ -369,4 +369,9 @@ def initialize_dummy_weights(
     """
     for param in model.state_dict().values():
         if torch.is_floating_point(param):
-            param.data.uniform_(low, high)
+            if torch.finfo(param.data.dtype).bits < 16:
+                # uniform_ doesn't support < 16-bit datatypes (FP8)
+                dtype = param.data.dtype
+                param.data.to(torch.float16).uniform_(low, high).to(dtype)
```

Comment:
I don't think this is doing what you want it to do -- it creates an FP16 copy, which is then modified. But the original tensor is kept unchanged:
```python
import torch

z = torch.zeros(3)

z.to(torch.float16).uniform_(-0.1, 1.0)

In [31]: z
Out[31]: tensor([0., 0., 0.])
```

In order to have this take effect, I believe you would need to assign it `param.data` at the end I believe :)

Reply:
You're right, I've missed that. fixed the PR, now it assigns the copy's data back to`param.data` .